app/tests/e2e/test_proof.py

```diff
@@ -0,0 +1,287 @@
+    async_client_alice_module_scope: AsyncClient,
+):
+    """this fixture produces the CRED_X_ID but if the test that produces the CRED_X_ID has already run
+    then this fixture just returns it..."""
+    credential = {
+        "protocol_version": "v1",
+        "connection_id": bob_connection_id,
+        "schema_id": schema_definition["schema_id"],
+        "attributes": {"speed": "average"},
+    }
+
+    response = await async_client_bob_module_scope.post(
+        "/generic/issuer/credentials",
+        data=json.dumps(credential),
+    )
+    credential_exchange = response.json()
+    credential_exchange_id = credential_exchange["credential_id"]
+    assert credential_exchange["protocol_version"] == "v1"
+
+    response = await async_client_alice_module_scope.get(
+        BASE_PATH, params={"connection_id": alice_connection_id}
+    )
+    records = response.json()
+    assert len(records) > 0
+
+    return credential_exchange_id
+
+
+proof_dict = dict(
+    {
+        "connection_id": "string",
+        "proof_request": {
+            "name": "string",
+            "non_revoked": {"from_": 0, "to": 0},
+            "nonce": "12345",
+            "requested_attributes": {
+                "0_string_uuid": {
+                    "name": "string",
+                    "names": ["string"],
+                    "non_revoked": {"from_": 0, "to": 0},
+                    "restrictions": None,
+                },
+            },
+            "requested_predicates": {
+                "0_string_GE_uuid": {
+                    "name": "string",
+                    "p_type": "<",
+                    "p_value": 0,
+                    "non_revoked": {"from_": 0, "to": 0},
+                    "restrictions": None,
+                },
+            },
+            "version": "0.1",
+        },
+        "comment": "string",
+        "trace": True,
+    }
+)
+
+
+@pytest.mark.asyncio
+async def test_send_proof_request(
+    alice_connection_id: str,
+    async_client_alice_module_scope: AsyncClient,
+):
+    # V1
+    response = await async_client_alice_module_scope.post(
+        BASE_PATH + f"/send-request?connection_id={alice_connection_id}",
+        data=json.dumps(proof_dict),
+    )
+
+    result = response.json()
+    assert "auto_present" is not None
+    assert "presentation" in result.keys()
+    assert "created_at" in result.keys()
+    assert "presentation_exchange_id" in result.keys()
+    assert result["role"] == "verifier"
+    assert result["state"]
+
+    # V2
+    response = await async_client_alice_module_scope.post(
+        BASE_PATH
+        + f"/send-request?connection_id={alice_connection_id}&protocol_version=2",
+        data=json.dumps(proof_dict),
+    )
+
+    result = response.json()
+    assert "auto_present" is not None
+    assert "presentation" in result.keys()
+    assert "created_at" in result.keys()
+    assert "presentation_exchange_id" in result.keys()
+    assert "v2-" in result["presentation_exchange_id"]
```

Comment:
In the issuer implementation I used `credential_id` insted of `credential_exchange_id`. Maybe we can here also use `proof_id` (or whather we are going to  call it instead of proof) instead of `presentation_exchange_id`. This feels more in line with the rest of the api